pkg/parser/tac/parse.go

```diff
@@ -0,0 +1,183 @@
+	}
+
+	return codes, nil
+}
+
+// Compare a struct's field names to the columns retrieved from the .txt file
+func ensureFileStructMatchesColumnNames(columnNames []string) error {
+	if len(columnNames) == 0 {
+		return errors.New("column names were not parsed properly from the second line of the tac file")
+	}
+	expectedColumnNames := getFieldNames(models.TransportationAccountingCodeTrdmFileRecord{})
+	if !reflect.DeepEqual(columnNames, expectedColumnNames) {
+		return errors.New("column names parsed do not match the expected format of tac file records")
+	}
+	return nil
+}
+
+// This function gathers the struct field names for comparison to
+// line 2 of the .txt file - The columns
+func getFieldNames(obj interface{}) []string {
+	var fieldNames []string
+
+	t := reflect.TypeOf(obj)
+	for i := 0; i < t.NumField(); i++ {
+		field := t.Field(i)
+		fieldNames = append(fieldNames, field.Name)
+	}
+
+	return fieldNames
+}
+
+// Removes all TACs with an expiration date in the past
+func PruneExpiredTACsDesiredFromTRDM(codes []models.TransportationAccountingCodeDesiredFromTRDM) []models.TransportationAccountingCodeDesiredFromTRDM {
+	var pruned []models.TransportationAccountingCodeDesiredFromTRDM
+
+	// If the expiration date is not before time.Now(), then it is not expired and should be appended to the pruned array
+	for _, code := range codes {
+		if !code.ExpirationDate.Before(time.Now()) {
+			pruned = append(pruned, code)
+		}
+	}
+
+	return pruned
+}
+
+// Consolidates TACs with the same TAC value. Duplicate "Transaction", aka description, calues are combined with a delimeter of ". Additional description found: "
+func ConsolidateDuplicateTACsDesiredFromTRDM(codes []models.TransportationAccountingCodeDesiredFromTRDM) []models.TransportationAccountingCodeDesiredFromTRDM {
+	consolidatedMap := make(map[string]models.TransportationAccountingCodeDesiredFromTRDM)
+
+	for _, code := range codes {
+		consolidatedMap[code.TAC] = overwriteDuplicateCode(consolidatedMap[code.TAC], code)
+	}
+
+	var consolidated []models.TransportationAccountingCodeDesiredFromTRDM
+	for _, value := range consolidatedMap {
+		consolidated = append(consolidated, value)
+	}
+
+	return consolidated
+}
+
+// This function checks two TAC codes: one existing and one new to decide which one to keep based on their ExpirationDates
+// If the ExpirationDate is the same, it appends the transactions and maintains the first code found
+// If the ExpirationDate is different, it appends the transactions and maintains the ExpirationDate further in the future
+func overwriteDuplicateCode(existingCode models.TransportationAccountingCodeDesiredFromTRDM, newCode models.TransportationAccountingCodeDesiredFromTRDM) models.TransportationAccountingCodeDesiredFromTRDM {
+
+	// If the new code expires later, append its transaction to the existing one (if not empty), and keep the new code
+	if newCode.ExpirationDate.After(existingCode.ExpirationDate) {
+		if existingCode.Transaction != "" {
```

Comment:
for simplicity, could these if statements be combined into 1

Reply:
Not quite for this spot. If the existing code has a blank transaction value, having the nested if statement prevents the delimiter ". Additional description found: " from being at the start of the new code's transaction

Reply:
ah.. good point, thnks for not just making the change based on suggestion